libs/sdk-ui-charts/src/highcharts/chartTypes/warterfallChart/waterfallChartsSeries.ts

```diff
@@ -0,0 +1,101 @@
+// (C) 2023 GoodData Corporation
+import { IMeasureGroupDescriptor } from "@gooddata/sdk-model";
+import { DataViewFacade } from "@gooddata/sdk-ui";
+import { IColorStrategy, valueWithEmptyHandling } from "@gooddata/sdk-ui-vis-commons";
+
+import { isOneOfTypes, parseValue, unwrap } from "../_util/common";
+import { multiMeasuresAlternatingTypes } from "../_chartOptions/chartCapabilities";
+import { IUnwrappedAttributeHeadersWithItems } from "../../typings/mess";
+import { IPointData, ISeriesItemConfig } from "../../typings/unsafe";
+
+function getSeriesItemData(
+    seriesItem: string[],
+    seriesIndex: number,
+    measureGroup: IMeasureGroupDescriptor["measureGroupHeader"],
+    viewByAttribute: IUnwrappedAttributeHeadersWithItems,
+    colorStrategy: IColorStrategy,
+    type: string,
+    emptyHeaderTitle: string,
+): IPointData[] {
+    return seriesItem.map((pointValue: string, pointIndex: number) => {
+        // by default seriesIndex corresponds to measureGroup label index
+        let measureIndex = seriesIndex;
+
+        if (isOneOfTypes(type, multiMeasuresAlternatingTypes) && !viewByAttribute) {
+            measureIndex = pointIndex;
+        }
+
+        const pointData: IPointData = Object.assign(
```

Comment:
Why use `Object.assign`? You can use spread operator. You will save lots of lines and it will be more align with how we code in these scenarios.

````
const marker = pointValue === null ? { marker: { enabled: false}} : {};
return {
   ...marker,
  y: parseValue(pointValue),
 format: unwrap(measureGroup.items[measureIndex]).format,
 name: <value>,
 legendIndex: <value>,
 index: <value>,
}